client/header/activity-panel/unread-indicators.js

```diff
@@ -57,3 +72,71 @@ export function getUnreadNotes( select ) {
 export function getLowStockCount() {
 	return getSetting( 'lowStockCount', 0 );
 }
+
+function getIncompleteTasksCount( tasks, dismissedTasks ) {
+	if ( ! tasks ) {
+		return 0;
+	}
+	return tasks.filter(
+		( task ) =>
+			task.visible &&
+			! task.completed &&
+			! dismissedTasks.includes( task.key )
+	).length;
+}
+
+function getAbbreviatedNotifications( select ) {
+	const { getOption } = select( OPTIONS_STORE_NAME );
+	const { getNotes } = select( NOTES_STORE_NAME );
+	const storeAlerts = getNotes( ALERTS_QUERY );
+	const thingsToDoNext = applyFilters(
+		'woocommerce_admin_onboarding_task_list',
+		[]
+	);
```

Comment:
Is there another way to get these tasks? Applying this filter in multiple spots is a code smell to me. If it's not possible any other way, we need to at least pass in the same parameters to the filter (like [`query`](https://github.com/woocommerce/woocommerce-admin/blob/c863e5d5c508d3de023df42a44e95557d5297890/client/task-list/tasks.js#L262)).

Also it seems that perhaps plugins aren't being loaded on "embed" pages (like WooCommerce > Orders) .. so 3PD code that would filter in a task isn't running (see https://github.com/woocommerce/woocommerce-admin/pull/7017#issuecomment-847272797).

Reply:
> Is there another way to get these tasks?

I'm using the same way we use to get the extended tasks in `Things to do next`. Sending the same parameters (`query`) to the filter is a good idea. I added them in the commit c1ef806

> Also it seems that perhaps plugins aren't being loaded on "embed" pages (like WooCommerce > Orders) .. so 3PD code that would filter in a task isn't running

This makes sense. Even [in our example](https://github.com/woocommerce/woocommerce-admin/blob/main/docs/examples/extensions/add-task/woocommerce-admin-add-task-example.php#L17) that's being prevented.
I think that until we switch the way we handle the extended tasks to `SlotFill` we will continue having this kind of issue.